node_package/src/helpers/renderHelper.ts

```diff
@@ -0,0 +1,19 @@
+import supportsReactCreateRoot from './supportsReactCreateRoot';
+import * as legacyRootHandler from './legacyRootHandler';
+import * as modernRootHandler from './modernRootHandler';
+
+
+let toExport;
+
+if (supportsReactCreateRoot) {
```

Comment:
This could be good based not only on React version but also some configuration option so it's an opt-in